src/content/blog/ddev-diffy-introduction.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Diffy integration with DDEV"
+pubDate: 2024-08-03
+modifiedDate: 2024-08-03
+summary: My experience building an addon for DDEV
+author: Yuri Gerasymov
+featureImage:
+  src: /img/blog/2024/08/ddev+diffy.png
+  alt: DDEV and Diffy integration
+  credit: 'Ideogram.ai: the words "2024" and "DDEV" next to each other'
+categories:
+  - Announcements
+---
+
+[Diffy](https://diffy.websote) is a visual regression testing tool that allows you to take screenshots of your website and compare them so you know exactly what pages got changed and where. We decided to build DDEV integration so it is possible to take screenshots from the local website, upload them to Diffy, and then compare them with screenshots from other environments.
+
+Watch [3 mins video](https://www.loom.com/share/1944eb3462934279a175e65e16e715ed) or read [documentation](https://docs.diffy.website/features/local-development/ddev-add-on).
+
+Architecturally integration consists of:
+
+* Docker container with preinstalled node and Chrome 111
+* Nodejs app that runs Chrome
+* The app itself is hosted in a separate repository as it is used in other places (like production for the main service itself).
+
+Let’s go through the way integration is built.
+
+## Download node app
+To download the app we use `install.yaml` post_install_actions:
+```
+- wget https://github.com/DiffyWebsite/diffy-worker/archive/refs/heads/main.zip
+- unzip main.zip
+- rm -f main.zip
+```
+
+## App configuration
+App requires an API key from the service and also project ID. This is a pretty interesting setup that we borrowed from the Platformsh project.
+We keep the API key in the global configuration (i.e. `~/.ddev/global_config.yaml`) and project ID is in local configuration (your project’s config.yaml).
+
+First we check if variables are set and if they are not — we request the values (`read token`) and then save them with `ddev config global —web-environment-add` in case of global variable and `ddev config —web-environment-add` in case of project based variable.
```

Comment:
```suggestion
First we check if variables are set and if they are not — we request the values (`read token`) and then save them with `ddev config global --web-environment-add` in case of global variable and `ddev config --web-environment-add` in case of project based variable.
```